base/strings/tagged.jl

```diff
@@ -0,0 +1,388 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+"""
+    TaggedString{S <: AbstractString} <: AbstractString
+
+A string with metadata, in the form of annotated regions.
+
+More specifically, this is a simple wrapper around any other
+[`AbstractString`](@ref) that allows for regions of the wrapped string to be
+annotated with tagged values.
+
+```text
+                        C
+                    ┌───┸─────────┐
+  "this is an example tagged string"
+  └──┰────────┼─────┘      │
+     A        └───┰────────┘
+                  B
+```
+
+Each tagged value (`A`, `B`, and `C` in the above diagram) is applied to a
+`UnitRange` of a string consists of a `Pair{Symbol, <:Any}` where the `Symbol`
+"tag" is used to label `Any` "value".
```

Comment:
This sentence probably needs to be reworded, right? I'm not able to follow quite exactly what it is trying to say here.

Reply:
Ah, yea. Does an "and" help much?

> Each tagged value (`A`, `B`, and `C` in the above diagram) is applied to a
`UnitRange` of the string, and consists of a `Pair{Symbol, <:Any}` where the `Symbol`
"tag" is used to label `Any` "value".

Reply:
Alternative version:

> The above diagram represents a `TaggedString` where three ranges have been
annotated (labeled `A`, `B`, and `C`). Each annotation must take the form of a
`Pair{Symbol, <:Any}`, where a `Symbol` "tag" is used to label `Any` "value".

Reply:
Used the alternative version.